### ISSUE TITLE
[Java] Feature request: google-api-client client library

### ISSUE THREAD

**Issue by charlescapps** [swagger-api/swagger-codegen#6825]
##### Description

The `google-api-client` [HTTP client](https://github.com/google/google-api-java-client) is a really standard HTTP client used in GCP, and it was used by a Google technology that was basically a clone of swagger (Cloud Endpoints). It's a very common HTTP framework used in GCP, and has good interop with the authentication abstractions provided by Google.

e.g. you can combine HTTPRequestInitializers (`com.google.api.client.http.HttpRequestInitializer`), and Google provides some nice classes that extend `HttpRequestInitializer` such as
* `com.google.api.client.auth.oauth2.Credential` - for OAuth authentication, which handles refreshing tokens and so on
* `com.google.api.client.googleapis.auth.oauth2.GoogleCredential` - which handles service account auth and also refreshes the token automagically.

TL; DR - we really want codegen for the `google-api-client` HTTP library!!

##### Swagger-codegen version

2.2.3. There's no support for this http library, it's not a regression.

##### Swagger declaration file content or url

n/a not a bug.

##### Command line used for generation

After we implement this, I would suggest the `library` is called `google-api-client`.

##### Steps to reproduce

n/a it's a new feature request

##### Related issues/PRs

I don't see anything related. Searching for "google-api-client" turns up 0 results in the issues.

##### Suggest a fix/enhancement

I'm planning to fix this by adding a new library type to the Java client codegen! My PR will be ready soon!
